### PR TITLE
Add Zabha extension

### PR DESCRIPTION
**Pull request by ved-rivos**
Specification: https://github.com/riscv/riscv-zabha

Tested with RISC-V arch. tests and checked against spike.

### CODE REVIEW

File: model/riscv_insts_aext.sail [hunk 1]
```diff
@@ -29,23 +29,30 @@ function lrsc_width_str(width : word_width) -> string =
...
  */
-function amo_width_valid(size : word_width) -> bool = {
+function lrsc_width_valid(size : word_width) -> bool = {
   match(size) {
     WORD   => true,
     DOUBLE => sizeof(xlen) >= 64,
     _      => false
   }
 }
+function amo_width_valid(size : word_width) -> bool = {
```
Comment (Timmmm) [inline model/riscv_insts_aext.sail:42]: Can you just add a blank line between these functions? Otherwise this LGTM.

**Review by Timmmm — APPROVED** [review]
LGTM

**Comment by Timmmm** [comment]
@billmcspadden-riscv any objection to merging this? I can't see any reason not to.

File: model/riscv_insts_aext.sail [hunk 2]
```diff
@@ -29,23 +29,31 @@ function lrsc_width_str(width : word_width) -> string =
...
+function lrsc_width_valid(size : word_width) -> bool = {
   match(size) {
     WORD   => true,
     DOUBLE => sizeof(xlen) >= 64,
     _      => false
   }
 }
 
+function amo_width_valid(size : word_width) -> bool = {
+  match(size) {
```
Comment (jrtc27) [inline model/riscv_insts_aext.sail:44]: Nit: the style of the existing amo_width_valid was wrong, there should be a space after `match`, and the parens also aren't needed (though that mistake is a little more common in the model). As you're copying this code could you please fix it first so we don't propagate more instances of the same mistake?

Comment (ved-rivos) [reply model/riscv_insts_aext.sail]: updated.

**Comment by Timmmm** [comment]
I think this is good to go. I'll merge this in a few days if nobody objects.

**Review by Alasdair — APPROVED** [review]

**Review by arichardson — APPROVED** [review]

**Pull request merged**